tests/unit/test_util.py

```diff
@@ -219,13 +222,20 @@ def test_sample_batch_indices_inter_row_uniqueness(self) -> None:
         batch_size = 200
         num_batches = 100
         batch_indices = sample_batch_indices(
-            random_key=key(0),
+            random_key=jr.key(0),
             max_index=max_index,
             batch_size=batch_size,
             num_batches=num_batches,
         )
         assert jnp.unique(batch_indices, axis=0).shape[0] == num_batches
 
+    def test_format_time(self) -> None:
+        """Test that `format_time` outputs expected strings."""
+        assert format_time(0.0004531) == "453.1 \u03bcs"
+        assert format_time(-0.032) == "-32.0 ms"
+        assert format_time(125) == "2.08 mins"
+        assert format_time(1e-15) == "0.0 ps"
```

Comment:
Good work incorporating an edge case. However, I would include at least one example for each unit, i.e. I'd like to see `0 s`, `1 s` and `1 ns` examples too.